.clang-tidy

```diff
@@ -38,30 +36,47 @@ Checks: >
     readability-*,
     -modernize-avoid-c-arrays,
     -modernize-use-bool-literals,
-    -modernize-use-auto
+    -modernize-use-auto,
+    -modernize-use-trailing-return-type,
+    -readability-magic-numbers,
+    -readability-implicit-bool-conversion,
+    -modernize-return-braced-init-list,
+    -readability-named-parameter,
+    -modernize-pass-by-value,
+    -modernize-use-default-member-init,
+    -bugprone-exception-escape,
+    -misc-non-private-member-variables-in-classes,
+    -misc-unused-parameters,
+    -readability-redundant-member-init,
+    -modernize-deprecated-headers,
+    -readability-identifier-naming,
```

Comment:
I can see that you have disabled many checks here, which is fine (if we can't do otherwise for now). I think it would be nice to add a few lines about why each check has been disabled (as I started to do above) in order to make it crystal clear that we have a reason to remove the checks.

Eg. If we have a lot of magic numbers (in the tests for eg) we can add a line to say "-readability-magic-numbers is removed because we have a lot of magic numbers in the tests [...]". Likewise, I imagine that you removed `readability-implicit-bool-conversion` because of the boolean vectors we use in the hash implementation, right?

Reply:
Added some comments about the disabled checks.